### ISSUE TITLE
Wandboard on kernel 3.18.0 vlan do not work

### ISSUE THREAD

**Issue by ghost** [RobertCNelson/armv7-multiplatform#23]
I am using Arch Linux ARM with kernel linux-armv7. Since I updated to 3.18.0 vlan do not work, no ping to machines on tagged vlan. Downgrading to 3.17.4 vlan work again.

On dmesg output all seems ok:

```
    $ dmesg | grep -i vlan
    [   19.818015] 8021q: 802.1Q VLAN Support v1.8
    [   20.184666] eth0: renamed from vlan10
```

**Comment by ghost** [comment]
After talk with fec kernel driver developers seems solved in linux-next:

> The issue must be fixed in linux-next by patch:
> 
> net: fec: init maximum receive buffer size for ring1 and ring2

I have not tested by now.

**Comment by ghost** [comment]
Confirmed, solved at linux-armv7-rc 4.0-rc5
